lib/recurly/resource.rb

```diff
@@ -1014,6 +1015,7 @@ def marshal_load(serialization)
         @destroyed,
         @uri,
         @href,
+        @type,
```

Comment:
Similarly, `@type` needs to be moved after `links` in this method. The reason for this is that if a user marshalled something with an old version of the library and then upgrades to a version that includes this change, it will assign the wrong data to the attribute. If we put it at the end of the array, it will just be nil if it's missing from the array altogether ensuring compatibility between versions of the library.

It won't look as pretty in the code, but it will be safer.